### PR TITLE
Autofill user location and weekday on index page

### PR DESCRIPTION
**Pull request by weihesdlegend**
## Description
Add button to locate user and autofill fields in search page

## Motivation
Users can find things to do in their current location with one click.

## Solution
JS built-in geolocation functionalities

## Testing
Manual tests

### CODE REVIEW

File: iowrappers/maps_client.go [hunk 1]
```diff
@@ -62,3 +62,23 @@ func CreateLogger() error {
...
 }
+
+func (mapsClient *MapsClient) ReverseGeocoding(context context.Context, latitude, longitude float64) (GeocodeQuery, error) {
+	request := &maps.GeocodingRequest{
+		LatLng: &maps.LatLng{
+			Lat: latitude,
+			Lng: longitude,
+		},
+		// currently we only need country and city info
+		ResultType: []string{"country", "locality"},
```
Comment (sananand007) [inline iowrappers/maps_client.go:73]: What is Locality? or is `city` called `locality` ?

File: templates/index.html [hunk 2]
```diff
@@ -30,10 +30,15 @@ <h1 style="color: #24c1e0; text-align: center; background-color: whitesmoke">
...
-        Please enter the city, country and weekday you want to visit
-    </h5>
-    <div>
+    <div class="column">
+        <h5 style="color: #24c1e0">
+            Please enter the city, country and weekday you want to visit
+        </h5>
+    </div>
+    <div class="column">
+        <button class="btn btn-outline-info btn-sm" id="find-my-location"> Find my location</button>
```
Comment (sananand007) [inline templates/index.html:39]: It might make more sense to say "Autofill current Location "  or just `Autofill` ?

File: templates/index.html [hunk 3]
```diff
@@ -76,6 +81,43 @@ <h5 style="color: #24c1e0">
...
 <script>
+    function locateMe() {
+        async function success(location) {
+            const latitude = location.coords.latitude;
+            const longitude = location.coords.longitude;
+            const date = new Date(location.timestamp);
+
+            console.log(latitude, longitude);
+
+            await fetch("/v1/reverse-geocoding" + "?lat=" + latitude.toString() + "&lng=" + longitude.toString())
```
Comment (sananand007) [inline templates/index.html:92]: Nice !!

**Review by sananand007 — COMMENTED** [review]
- Can we push the filling fields down a bit and have the "Autofill" button on the center? I think it might look cleaner that way ?

File: iowrappers/maps_client.go (hunk 1 above)
Comment (weihesdlegend) [reply iowrappers/maps_client.go:73]: city called locality

File: templates/index.html (hunk 2 above)
Comment (weihesdlegend) [reply templates/index.html:39]: I think `Autofill` makes sense

**Review by sananand007 — APPROVED** [review]
LGTM , The Autofill is left-aligned, but still looks good.

**Pull request merged**